libs/tendermint/state/txindex/indexer_service.go

```diff
@@ -51,28 +51,33 @@ func (is *IndexerService) OnStart() error {
 	}
 
 	go func() {
-		for is.IsRunning() {
```

Comment:
Can delete isRunning, and always use case <-blockHeadersSub.Cancelled():

Reply:
good advice